pkg/webhook/preflight/nutanix/failuredomain.go

```diff
@@ -0,0 +1,205 @@
+		for _, fdName := range cd.nutanixClusterConfigSpec.ControlPlane.Nutanix.FailureDomains {
+			if fdName != "" {
+				checks = append(checks, &failureDomainCheck{
+					failureDomainName: fdName,
+					namespace:         cd.cluster.Namespace,
+					field: fmt.Sprintf(
+						"cluster.spec.topology.variables[.name=clusterConfig].value.controlPlane.nutanix.failureDomains[%s]",
+						fdName,
+					),
+					kclient: cd.kclient,
+					nclient: cd.nclient,
+				})
+			}
+		}
+	}
+
+	// For the failure domains configured for worker nodes
+	if cd.cluster != nil &&
+		cd.cluster.Spec.Topology != nil &&
+		cd.cluster.Spec.Topology.Workers != nil {
+		for i := range cd.cluster.Spec.Topology.Workers.MachineDeployments {
+			md := &cd.cluster.Spec.Topology.Workers.MachineDeployments[i]
+			if md.FailureDomain != nil && *md.FailureDomain != "" {
+				checks = append(checks, &failureDomainCheck{
+					failureDomainName: *md.FailureDomain,
+					namespace:         cd.cluster.Namespace,
+					field: fmt.Sprintf(
+						"cluster.spec.topology.workers.machineDeployments[.name=%s].failureDomain[%s]",
+						md.Name,
+						*md.FailureDomain,
+					),
+					kclient: cd.kclient,
+					nclient: cd.nclient,
+				})
+			}
+		}
+	}
+
+	return checks
+}
+
+func (fdc *failureDomainCheck) Run(ctx context.Context) preflight.CheckResult {
+	result := preflight.CheckResult{
+		Allowed: true,
+	}
+
+	// Fetch the referent failure domain object
+	fdObj := &capxv1.NutanixFailureDomain{}
+	fdKey := ctrlclient.ObjectKey{Name: fdc.failureDomainName, Namespace: fdc.namespace}
+	if err := fdc.kclient.Get(ctx, fdKey, fdObj); err != nil {
+		msg := ""
+		if errors.IsNotFound(err) {
+			msg = fmt.Sprintf("not found the failure domain object with name %q", fdc.failureDomainName)
+		} else {
+			msg = fmt.Sprintf("failed to fetch the failure domain object with name %q: %v", fdc.failureDomainName, err)
+		}
+
+		result.Allowed = false
+		result.Causes = append(result.Causes, preflight.Cause{
+			Message: msg,
+			Field:   fdc.field,
+		})
+
+		return result
+	}
+
+	// Validate the failure domain configuration
+	// Validate spec.prismElementCluster configuration
+	peIdentifier := fdObj.Spec.PrismElementCluster
+	peClusters, err := getClusters(fdc.nclient, &peIdentifier)
+	if err != nil {
+		result.Allowed = false
+		result.InternalError = true
+		result.Causes = append(result.Causes, preflight.Cause{
+			Message: fmt.Sprintf(
+				"failed to get the prism element cluster %s configured in the failure domain %q: %v",
+				peIdentifier,
+				fdc.failureDomainName,
+				err,
+			),
+			Field: fmt.Sprintf("%s: .spec.prismElementCluster", fdc.field),
+		})
+		return result
+	}
+	if len(peClusters) != 1 {
+		result.Allowed = false
+		result.Causes = append(result.Causes, preflight.Cause{
+			Message: fmt.Sprintf(
+				"expected to find 1 prism element cluster %s, found %d",
+				peIdentifier,
+				len(peClusters),
+			),
+			Field: fmt.Sprintf("%s: .spec.prismElementCluster", fdc.field),
```

Comment:
We want to use valid field syntax, and since the preflight check is in the context of the Cluster resource, we should use a field from the Cluster.

```suggestion
			Field: fdc.field,
```

Reply:
changed the field as suggested